server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Scanner.java

```diff
@@ -38,100 +40,126 @@
   private ScanDataSource isolatedDataSource;
   private boolean sawException = false;
   private boolean scanClosed = false;
+  protected Semaphore scannerSemaphore;
 
   Scanner(Tablet tablet, Range range, ScanOptions options) {
     this.tablet = tablet;
     this.range = range;
     this.options = options;
+    scannerSemaphore = new Semaphore(1, true);
```

Comment:
This semaphore replaces a synchronized block with a fair semaphore.  AFAICT from searching around on the web, synchronized blocks are not fair.   Do we need fairness here?  The javadocs say its more expensive.   If its not needed, not sure should set it.

I was trying to determine the difference between the semaphore and a lock for this case.  It seems like the only difference is re-entrance, which it does not seem is needed in this case.